pkg/v1/cli/discovery/kubernetes_test.go

```diff
@@ -0,0 +1,100 @@
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/cli/plugin"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/fakes"
+	fakehelper "github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/fakes/helper"
+)
+
+func TestClient(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Discovery Suite")
+}
+
+var _ = Describe("Unit tests for kubernetes discovery", func() {
+	var (
+		err                  error
+		currentClusterClient *fakes.ClusterClient
+		kd                   *discovery.KubernetesDiscovery
+		plugins              []plugin.Discovered
+		cliplugins           []v1alpha1.CLIPlugin
+	)
+
+	Describe("When Getting Discovered plugins from k8s cluster", func() {
+		BeforeEach(func() {
+			currentClusterClient = &fakes.ClusterClient{}
+			kd = &discovery.KubernetesDiscovery{}
+		})
+
+		JustBeforeEach(func() {
+			plugins, err = kd.GetDiscoveredPlugins(currentClusterClient)
+		})
+
+		Context("When CLIPlugin CRD verification throws error", func() {
+			BeforeEach(func() {
+				currentClusterClient.VerifyCLIPluginCRDReturns(false, errors.New("fake error"))
+			})
+			It("return empty plugin list without an error", func() {
+				Expect(len(plugins)).To(Equal(0))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+		Context("When CLIPlugin CRD verification returns false and no error", func() {
+			BeforeEach(func() {
+				currentClusterClient.VerifyCLIPluginCRDReturns(false, nil)
+			})
+			It("return empty plugin list without an error", func() {
+				Expect(len(plugins)).To(Equal(0))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+		Context("When CLIPlugin CRD verification returns true with error", func() {
+			BeforeEach(func() {
+				currentClusterClient.VerifyCLIPluginCRDReturns(true, errors.New("fake error"))
+			})
+			It("return empty plugin list without an error", func() {
+				Expect(len(plugins)).To(Equal(0))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Context("When ListCLIPluginResources returns error", func() {
+			BeforeEach(func() {
+				currentClusterClient.VerifyCLIPluginCRDReturns(true, nil)
+				currentClusterClient.ListCLIPluginResourcesReturns(nil, errors.New("fake error"))
+			})
+			It("return empty plugin list with an error", func() {
+				Expect(len(plugins)).To(Equal(0))
+				Expect(err).To(HaveOccurred())
+			})
+		})
+		Context("When ListCLIPluginResources list of CLIPlugin resources", func() {
+			BeforeEach(func() {
+				cliplugin1 := fakehelper.NewCLIPlugin(fakehelper.TestCLIPluginOption{Name: "plugin1", Description: "plugin1 desc", RecommendedVersion: "v0.0.1"})
+				cliplugin2 := fakehelper.NewCLIPlugin(fakehelper.TestCLIPluginOption{Name: "plugin2", Description: "plugin2 desc", RecommendedVersion: "v0.0.2"})
+				cliplugins = append(cliplugins, cliplugin1, cliplugin2)
+				currentClusterClient.VerifyCLIPluginCRDReturns(true, nil)
+				currentClusterClient.ListCLIPluginResourcesReturns(cliplugins, nil)
+			})
+			It("return list of plugins without error", func() {
```

Comment:
nit: It looks like we're returning an ordered list. Would it help to add that in the description of the test being run here? i.e `It("return ordered list...)`.